src/app/shared/directives/email.directive.ts

```diff
@@ -1,13 +1,15 @@
 import { Directive } from "@angular/core";
-import { AbstractControl, ValidationErrors, Validator } from "@angular/forms";
+import { AbstractControl, ValidationErrors, Validator, NG_VALIDATORS } from "@angular/forms";
 
 @Directive({
     selector: '[emailValidator]',
-    providers: [/*Add your code here*/]
+    providers: [{ provide: NG_VALIDATORS, useExisting: EmailValidatorDirective, multi: true }]
```

Comment:
providers обыно бьем на несколько строчек

```suggestion
    providers: [
      {
        provide: NG_VALIDATORS,
        useExisting: EmailValidatorDirective,
        multi: true
      }
    ]
```